packages/ckeditor5-list/src/documentlist/utils/listwalker.js

```diff
@@ -95,6 +121,13 @@ export default class ListWalker {
 		 */
 		this._sameItemType = !!options.sameItemType;
 
+		/**
+		 * Whether should return only blocks with the same `listStyle`, `listReversed` and `listStart` attributes.
+		 * @private
+		 * @type {Boolean}
+		 */
+		this._sameProperties = !!options.sameProperties;
+
```

Comment:
Those properties are optional feature-related and extend the set of base list attributes (indent, type, item id). I think that there is no point in extending ListWalker with additional attributes, especially since you could stop iterating it as soon as you encounter the abort element from outside of the walker itself.